test/framework/kogitodataindex.go

```diff
@@ -15,94 +15,31 @@
 package framework
 
 import (
-	"fmt"
-	"strconv"
-
-	apps "k8s.io/api/apps/v1"
-
 	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
 	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
 	"github.com/kiegroup/kogito-cloud-operator/test/config"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// InstallKogitoDataIndexService deploy the Kogito Data Index service
+// InstallKogitoDataIndexService install the Kogito Data Index service
 func InstallKogitoDataIndexService(namespace string, installerType InstallerType, replicas int) error {
-	GetLogger(namespace).Infof("%s install Kogito Data Index with %d replicas", installerType, replicas)
-	switch installerType {
-	case CLIInstallerType:
-		return cliInstallKogitoDataIndex(namespace, replicas)
-	case CRInstallerType:
-		return crInstallKogitoDataIndex(namespace, int32(replicas))
-	default:
-		panic(fmt.Errorf("Unknown installer type %s", installerType))
-	}
-}
-
-func crInstallKogitoDataIndex(namespace string, replicas int32) error {
-	kogitoDataIndex := &v1alpha1.KogitoDataIndex{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      infrastructure.DefaultDataIndexName,
-			Namespace: namespace,
-		},
+	resource := &v1alpha1.KogitoDataIndex{
+		ObjectMeta: NewObjectMetadata(namespace, getDataIndexServiceName()),
 		Spec: v1alpha1.KogitoDataIndexSpec{
-			KogitoServiceSpec: v1alpha1.KogitoServiceSpec{
-				Replicas: &replicas,
-				Image:    getDataIndexImage(),
-			},
+			KogitoServiceSpec: NewKogitoServiceSpec(int32(replicas), config.GetDataIndexImageTag(), infrastructure.DefaultDataIndexImageFullTag),
 		},
 		Status: v1alpha1.KogitoDataIndexStatus{
-			KogitoServiceStatus: v1alpha1.KogitoServiceStatus{
-				ConditionsMeta: v1alpha1.ConditionsMeta{
-					Conditions: []v1alpha1.Condition{},
-				},
-			},
+			KogitoServiceStatus: NewKogitoServiceStatus(),
 		},
 	}
 
-	if _, err := kubernetes.ResourceC(kubeClient).CreateIfNotExists(kogitoDataIndex); err != nil {
-		return fmt.Errorf("Error creating Kogito Data Index service: %v", err)
-	}
-	return nil
-}
-
-func cliInstallKogitoDataIndex(namespace string, replicas int) error {
-	cmd := []string{"install", "data-index"}
-	cmd = append(cmd, "--image", framework.ConvertImageToImageTag(getDataIndexImage()))
-	cmd = append(cmd, "--replicas", strconv.Itoa(replicas))
-
-	_, err := ExecuteCliCommandInNamespace(namespace, cmd...)
-	return err
-}
-
-func getDataIndexImage() v1alpha1.Image {
-	if len(config.GetDataIndexImageTag()) > 0 {
-		return framework.ConvertImageTagToImage(config.GetDataIndexImageTag())
-	}
-
-	image := framework.ConvertImageTagToImage(infrastructure.DefaultDataIndexImageFullTag)
-	image.Tag = config.GetServicesImageVersion()
-	return image
+	return InstallServiceWithoutPersistence(resource, installerType, "data-index")
```

Comment:
can you comment why you install without persistence ?

Reply:
I don't know what you mean. This should be used only when the services do not support persistence (the enable-persistence in CLI for example).

Reply:
just add a small comment because it can be confusing. data-index needs persistence per definition. but here we install it without.
just add a comment like `CLI for data-index does not support extra persistence installation` or similar

Reply:
Added